### PR TITLE
Update docs for #982

### PR DESCRIPTION
**Pull request by dialvarezs**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [ ] Have you got 100% test coverage on new code?
- [ ] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

This PR adds a new section in the DTO docs to reflect the automatic DTO conversion from response data implemented in #982.
Also extracts all the examples from the DTO section, and implement tests for them.
And fixes a typo in CONTRIBUTING.md

### CODE REVIEW

**Comment by Goldziher** [comment]
Thanks for this PR! Very nice addition. Before we merge it though - we are in the process of migrating all examples into the examples folder and testing them. You can see examples for this throughout the code base and especially read about it in the CONTRIBUTING.md file -> the `Adding Examples` section.

File: docs/usage/11-data-transfer-objects/0-dto.md [hunk 1]
```diff
@@ -260,3 +260,53 @@ In the above `company_instance` is an instance of the SQLAlchemy declarative cla
...
+        return companies[company_id - 1]
+    except IndexError:
+        raise HTTPException(
+            detail="Company not found",
+            status_code=HTTP_404_NOT_FOUND,
+        )
+
+
+@get()
+def get_companies() -> list[ReadCompanyDTO]:
```
Comment (provinzkraut) [inline docs/usage/11-data-transfer-objects/0-dto.md:308]: Please use Python 3.8 compatible type annotations.

**Review by provinzkraut — CHANGES_REQUESTED** [review]

**Comment by dialvarezs** [comment]
@Goldziher 
- Do I need to extract the smaller codes as well? (pydantic model definitions)
- Do I need to create a test for every example? In CONTRIBUTING.md says the tests should be in `tests/examples`, but that doesn't exist.

**Comment by Goldziher** [comment]
> @Goldziher
> 
> * Do I need to extract the smaller codes as well? (pydantic model definitions)
> * Do I need to create a test for every example? In CONTRIBUTING.md says the tests should be in `tests/examples`, but that doesn't exist.

so please fix CONTRIBUTING.md, its actually `examples/tests`.

You should test every example that is a runnable code snippet. Smaller examples that just show partial code do not need to be tested because they are untestable.

**Comment by dialvarezs** [comment]
I'm not sure how should I fix the remaining linting errors 🤔.

**Comment by Goldziher** [comment]
have you installed `pre-commit` locally? Once you have it, you can run `pre-commit run --all-files` and reproduce the errors locally.

**Comment by dialvarezs** [comment]
> have you installed `pre-commit` locally? Once you have it, you can run `pre-commit run --all-files` and reproduce the errors locally.

Yes, I did, but I'm talking about how to fix them. They are mostly about types. Should I ignore the checks for them?

**Comment by Goldziher** [comment]
> > have you installed `pre-commit` locally? Once you have it, you can run `pre-commit run --all-files` and reproduce the errors locally.
> 
> Yes, I did, but I'm talking about how to fix them. They are mostly about types. Should I ignore the checks for them?

yes, you can type ignore the errors

**Comment by dialvarezs** [comment]
It's ready then, I guess.

**Review by Goldziher — APPROVED** [review]
great stuff! thanks 😉

**Comment by Goldziher** [comment]
@all-contributors add @dialvarezs docs

**Pull request merged**